src/View/PropertySpace.java

```diff
@@ -0,0 +1,130 @@
+ * The PropertySpace class is an extensions of the JPanel and creates
+ * a functional tile for one space on the Monopoly Board. It holds
+ * functionality for tracking the name and price of a property as well
+ * as how many houses are on the space.
+ *
+ * @author David Baas
+ * @version 2/21/2018
+ *********************************************************************/
+public class PropertySpace extends JPanel {
+
+  /**A Label for the name of the property*/
+  private JLabel name;
+
+  /**A label for the price of the property*/
+  private JLabel price;
+
+  /**A Housing panel for drawing houses */
+  private HousingPanel housingPanel;
+
+  /********************************************************************
+   * The construtor initializes all of the variables and sets the
+   * color of the property.
+   * @param c The color of the property.
+   * @param name The name of the property.
+   * @param price The price of the property.
+   * @throws IOException If the images for the HousingPanel don't load.
+   *******************************************************************/
+  public PropertySpace(Color c,String name, int price) throws IOException{
+    this.name = new JLabel(name);
+    this.price = new JLabel(""+price);
+    housingPanel = new HousingPanel(c);
+
+    setTextAlignment();
+    createPanel();
+  }
+
+  /*******************************************************************
+   * The getHouses method returns the number of houses on this property.
+   *
+   * @return houseCounter The number houses on this property.
+   ******************************************************************/
+  public int getHouses() {
+    return housingPanel.getHouses();
+  }
+
+  /*******************************************************************
+   * The setHouses method sets the number of houses on this property.
+   * @param houseCounter the number of houses on this property.
+   ******************************************************************/
+  public void setHouses(int houseCounter) {
+    housingPanel.setHouses(houseCounter);
+  }
+
+  /********************************************************************
+   * The setTextAlignment method is a private method that aligns the
+   * text of the Labels to the center of the panel.
+   *******************************************************************/
+  private void setTextAlignment(){
+    name.setHorizontalTextPosition(SwingConstants.CENTER);
+    price.setHorizontalTextPosition(SwingConstants.CENTER);
+    price.setVerticalAlignment(SwingConstants.BOTTOM);
+  }
+
+  /********************************************************************
+   * The createPanel method is a private method that places the
+   * JComponents into the correct positions to resemble a Monopoly
+   * space.
+   *******************************************************************/
+  private void createPanel() {
+    setLayout(new GridBagLayout());
+    GridBagConstraints g = new GridBagConstraints();
```

Comment:
I'm not sure how to do it the best way, but depending on flow layout might help us relocate things in an easier manner when it comes to the tile.

Reply:
I tried every layout under the sun to make this work and the GridBagLayout was the only thing that made this look correct.

Reply:
I tried every layout under the sun to make this work and the GridBagLayout was the only thing that made this look correct.